fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadata.java

```diff
@@ -151,12 +189,17 @@ private void collectDeltaLakePlanFiles(PredicateSearchKey key, Table table, Scal
                 new ScalarOperationToDeltaLakeExpr.DeltaLakeContext(schema, partitionColumns);
         Predicate deltaLakePredicate = new ScalarOperationToDeltaLakeExpr().convert(scalarOperators, deltaLakeContext);
 
-        ScanBuilder scanBuilder = snapshot.getScanBuilder(engine);
-        Scan scan = scanBuilder.withFilter(engine, deltaLakePredicate).build();
+        ScanBuilderImpl scanBuilder = (ScanBuilderImpl) snapshot.getScanBuilder(engine);
+        ScanImpl scan = (ScanImpl) scanBuilder.withFilter(engine, deltaLakePredicate).build();
+
+        List<String> nonPartitionPrimitiveColumns = schema.fieldNames().stream()
+                .filter(column -> BasePrimitiveType.isPrimitiveType(
+                        schema.get(column).getDataType().toString())
+                        && !partitionColumns.contains(column)).collect(toImmutableList());
 
         List<Row> files = Lists.newArrayList();
 
-        try (CloseableIterator<FilteredColumnarBatch> scanFilesAsBatches = scan.getScanFiles(engine)) {
+        try (CloseableIterator<FilteredColumnarBatch> scanFilesAsBatches = scan.getScanFiles(engine, true)) {
```

Comment:
I think there no need to set the includeStats = true, delta sdk could include the stats for data skip column even if includeStats =false
![image](https://github.com/StarRocks/starrocks/assets/9495145/a3d04dbf-3026-4d5d-818c-83318b18f87b)

Reply:
private Optional<DataSkippingPredicate> getDataSkippingFilter() {
        return getDataFilters().flatMap(dataFilters ->
            DataSkippingUtils.constructDataSkippingFilter(dataFilters, metadata.getDataSchema())
        );
    }
 will remove always true. But optimizer need all the column statistics not only columns of predicate.